typescript/web/next-sitemap.js

```diff
@@ -0,0 +1,3 @@
+module.exports = {
+  siteUrl: "https://labelflow.ai",
+};
```

Comment:
This file shouldn't be required:
[![image](https://user-images.githubusercontent.com/2091902/148396281-9d080ca7-d6d1-4e93-84e5-001ffbcf549a.png)](https://www.npmjs.com/package/next-sitemap)

Also siteUrl can't be hardcoded to `https://labelflow.ai` it it can change depending on the content of the .env files and environment variables.

Reply:
It is required, it's written in the screenshot you attached.
If you don't put one you get the following error:
```
(node:69018) UnhandledPromiseRejectionWarning: Error: /Users/jordan/dev/labelflow/typescript/web/next-sitemap.js does not exist.
    at Object.exports.getConfigFilePath (/Users/jordan/dev/labelflow/node_modules/next-sitemap/dist/cjs/path/index.js:45:15)
    at /Users/jordan/dev/labelflow/node_modules/next-sitemap/dist/cjs/cli.js:22:35
    at Generator.next (<anonymous>)
    at /Users/jordan/dev/labelflow/node_modules/next-sitemap/dist/cjs/cli.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/Users/jordan/dev/labelflow/node_modules/next-sitemap/dist/cjs/cli.js:4:12)
    at /Users/jordan/dev/labelflow/node_modules/next-sitemap/dist/cjs/cli.js:20:8
    at Object.<anonymous> (/Users/jordan/dev/labelflow/node_modules/next-sitemap/dist/cjs/cli.js:50:4)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:69018) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:69018) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Regarding the siteUrl, do we already have an env variable for this ?

Reply:
Yes, I've read the doc but the message below would let us think that it's not necessary anymore 😉 

Actually, we should avoid relying on the public URL at build-time. This is something which is preferrable to allow being changed as close to runtime as possible.

Reply:
Is the URL really needed?

Reply:
> Actually, we should avoid relying on the public URL at build-time.

But `next-sitemap` is based on the output of `next-build` so why would we want to keep this flexibility ?

> Is the URL really needed?

Yes. If you don't fill it in, you will end up with this kind of output:
```xml
<url><loc>undefined/about</loc><changefreq>daily</changefreq><priority>0.7</priority><lastmod>2022-01-06T16:37:30.326Z</lastmod></url>
```

Where `undefined` is (should be) your siteUrl

Reply:
> keep this flexibility ?

It's not flexible if the URL is needed during the build process since, for example, we would end-up needing to create 1 docker image per-URL we'd like to use.

If we really can't do things at runtime (e.g. by calling an hypothetical next-sitemap lib which would generate the XML file with SSR), then the call to `next-sitemap` should be moved from the build phase to the deployment phase.

Reply:
I don't see the relation with Docker at this point 🤔

Reply:
Because, this would either means that customers would have our own URL in their sitemap or that we would be forced use to build a Docker image for each customer once #615.
This is the same problem I had with the IOG URL in https://github.com/labelflow/labelflow/pull/615#issuecomment-996526596